src/components/CreateGroup/CreateGroupForm.tsx

```diff
@@ -0,0 +1,107 @@
+import React from 'react'
+import Typography from '@material-ui/core/Typography'
+import Box from '@material-ui/core/Box'
+import { TextField } from 'formik-material-ui'
+import { spacing } from '@material-ui/system'
+import { styled } from '@material-ui/core/styles'
+import MuiButton from '@material-ui/core/Button'
+import { Formik, Form, Field } from 'formik'
+import CheckBoxIcon from '@material-ui/icons/CheckBox'
+import { useTranslation } from 'react-i18next'
+import { selectToken } from '../App/userSelector'
+import { useSelector } from 'react-redux'
+import { createGroup } from '../../pages/AdminDashboard/groupsActions'
+import { useDispatch } from 'react-redux'
+
+const Button = styled(MuiButton)(spacing)
+
+export interface Group {
+    name: string
+    password: string
+}
+
+const CreateGroupForm = () => {
+    const { t } = useTranslation('create-group-form')
+    const token = useSelector(selectToken)
+    const dispatch = useDispatch()
+
+
+    const validateForm = (values) => {
+        const errors: Partial<Group> = {}
+        const requiredFields = ['name', 'password']
+        for (const requiredField of requiredFields) {
+            if (!values[requiredField]) {
+                errors[requiredField] = 'Required'
+            }
+        }
+        return errors
+    }
+
+    return (
+        <Formik
+            initialValues={{
+                name: '',
+                password: '',
+            }}
+            validate={(values) => validateForm(values)}
+            onSubmit={(values, { setSubmitting }) => {
+                setSubmitting(true)
+                dispatch(createGroup(t,values))
+                setSubmitting(false)
+            }}>
+            {({ submitForm, isSubmitting, isValid }) => (
+                <Form>
+                    <Typography variant="h5" component="h1">
```

Comment:
why `variant=h5` while `component=h1` ?

Reply:
Because it's the main title of my component but I want it to be styled like a material h5. (like in room creation form)